mono/metadata/threads.c

```diff
@@ -1641,56 +1638,72 @@ ves_icall_System_Threading_Thread_GetName_internal (MonoInternalThread *this_obj
 }
 
 void 
-mono_thread_set_name_internal (MonoInternalThread *this_obj, MonoString *name, gboolean permanent, gboolean reset, MonoError *error)
+mono_thread_set_name_internal (MonoInternalThread *this_obj, const char* name_utf8, size_t length, gboolean permanent, gboolean reset, MonoError *error)
```

Comment:
We use `gsize` instead of `size_t`, and `gchar` instead of `char`

Reply:
Web says gchar is historical and pointless but ok. HavocP on stack overflow who works on this stuff I believe. Gsize seems similar.

Reply:
I thought I'd seen a fair amount of char & size_t & hadn't actually noticed gchar or gsize. I only noticed the stackoverflow stuff much later.

Reply:
$ pwd
/dev2/mono/mono

# false hits
$ git grep "signed char" |wc -l
     820

and then the g's are heavily used, but the minority:

$ git grep \\bchar\\b |wc -l
    7794

$ git grep \\bgchar\\b |wc -l
    1304

$ git grep \\bsize_t\\b |wc -l
    1180

$ git grep \\bgsize\\b |wc -l
     626

Reply:
I have no strong feeling on `char` vs `gchar` but I do think `gsize` is both nicer to read and write than `size_t`.